app/portainer/views/stacks/edit/stackController.js

```diff
@@ -229,6 +236,28 @@ angular.module('portainer.app').controller('StackController', [
       $scope.state.actionInProgress = false;
     }
 
+    function validateYAML(yaml) {
```

Comment:
this method is duplicate from previous one, maybe we should share it in a helper